src/Weapon.cpp

```diff
@@ -1,7 +1,21 @@
 #include "Weapon.h"
 
-Weapon::Weapon(){
-
+Weapon::Weapon(std::string type, 
```

Comment:
So close! you fixed all sub classes but this should use member initializers as well.